lib/Service/BreakoutRoomService.php

```diff
@@ -0,0 +1,237 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2022 Joas Schilling <coding@schilljs.com>
+ *
+ * @author Joas Schilling <coding@schilljs.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Service;
+
+use InvalidArgumentException;
+use OCA\Talk\Manager;
+use OCA\Talk\Model\Attendee;
+use OCA\Talk\Model\BreakoutRoom;
+use OCA\Talk\Participant;
+use OCA\Talk\Room;
+use OCA\Talk\Webinary;
+use OCP\EventDispatcher\IEventDispatcher;
+use OCP\IL10N;
+
+class BreakoutRoomService {
+	protected Manager $manager;
+	protected RoomService $roomService;
+	protected ParticipantService $participantService;
+	protected IEventDispatcher $dispatcher;
+	protected IL10N $l;
+
+	public function __construct(Manager $manager,
+								RoomService $roomService,
+								ParticipantService $participantService,
+								IEventDispatcher $dispatcher,
+								IL10N $l) {
+		$this->manager = $manager;
+		$this->roomService = $roomService;
+		$this->participantService = $participantService;
+		$this->dispatcher = $dispatcher;
+		$this->l = $l;
+	}
+
+	/**
+	 * @param Room $parent
+	 * @param int $mode
+	 * @psalm-param 0|1|2|3 $mode
+	 * @param int $amount
+	 * @param string $attendeeMap
+	 * @return Room[]
+	 * @throws InvalidArgumentException When the breakout rooms are configured already
+	 */
+	public function setupBreakoutRooms(Room $parent, int $mode, int $amount, string $attendeeMap): array {
```

Comment:
This var name will make the code more readable in all places of this class that use `$parent` as Room object.
```suggestion
	public function setupBreakoutRooms(Room $parentRoom, int $mode, int $amount, string $attendeeMap): array {
```

Reply:
I think it should be obvious enough as there is no other item that could be a parent. Also the type hint indicates it as well.
Also I want to avoid breakings now with the follow up pull requests.

Reply:
Is a good point to move to part 2.